Firestore/core/src/model/document_key.cc

```diff
@@ -49,14 +49,26 @@ DocumentKey::DocumentKey(ResourcePath&& path)
   AssertValidPath(*path_);
 }
 
+/* static */
 DocumentKey DocumentKey::FromPathString(const std::string& path) {
-  return DocumentKey{ResourcePath::FromStringView(path)};
+  return DocumentKey{ResourcePath::FromString(path)};
 }
 
+/* static */
 DocumentKey DocumentKey::FromSegments(std::initializer_list<std::string> list) {
   return DocumentKey{ResourcePath{list}};
 }
 
+/* static */
+DocumentKey DocumentKey::FromName(absl::string_view name) {
+  ResourcePath resource_name = ResourcePath::FromString(name);
+  HARD_ASSERT(resource_name.size() >= 4 && resource_name[0] == "projects" &&
+                  resource_name[2] == "databases" &&
+                  resource_name[4] == "documents",
```

Comment:
Since you're accessing the fifth element (`[4]`), the size check should be `> 4`, not `>= 4`.

Reply:
Fixed here (Android coming)